src/main/java/Storage/Storage.java

```diff
@@ -0,0 +1,104 @@
+package Storage;
+
+import Common.CommonFunctions;
+import Task.Task;
+import Task.Todo;
+import Task.Event;
+import Task.Deadline;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Storage {
+    private File dukeFile;
+
+    public Storage() {
+        createDataFolder();
+        createDukeFile();
+    }
+
+    private void createDataFolder() {
+        File dataFolder = new File("data");
+        if (!dataFolder.isDirectory()) {
+            try {
+                dataFolder.mkdir();
+            } catch (SecurityException e) {
+                CommonFunctions.printMessage(e.getMessage());
+            }
+        }
+    }
+
+    private void createDukeFile() {
+        File dukeFile = new File("data/Duke.txt");
+        if (!dukeFile.exists()) {
+            try {
+                dukeFile.createNewFile();
+            } catch (SecurityException e) {
```

Comment:
Can consider catching multiple exceptions in one catch block to simplify your code. http://tutorials.jenkov.com/java-exception-handling/catching-multiple-exceptions.html